docs/development/extensions-contrib/aliyun-oss-extensions.md

```diff
@@ -22,33 +22,215 @@ title: "Aliyun OSS"
   -->
 
+[Alibaba Cloud](https://www.aliyun.com) is the 3rd largest cloud infrastructure provider in the world. It provides its own storage solution known as OSS, [Object Storage Service](https://www.aliyun.com/product/oss).
+This document describes how to use OSS as Druid deep storage.
 
-To use this Apache Druid extension, make sure to [include](../../development/extensions.md#loading-extensions) `aliyun-oss-extensions` extension.
+## Installation
 
-## Deep Storage
+Use [pull-deps](../../operations/pull-deps.md) tool shipped with Druid to install this [extension](../../development/extensions.md#community-extensions) on middle manager and historical nodes.
 
-[Aliyun](https://www.aliyun.com) is the 3rd largest cloud infrastructure provider in the world. It provides its own storage solution known as OSS, [Object Storage Service](https://www.aliyun.com/product/oss). 
+```bash
+java -classpath "{YOUR_DRUID_DIR}/lib/*" org.apache.druid.cli.Main tools pull-deps -c org.apache.druid.extensions.contrib:aliyun-oss-extensions:{YOUR_DRUID_VERSION}
+```
 
-To use aliyun OSS as deep storage, first config as below
+## Enabling
 
-|Property|Description|Possible Values|Default|
-|--------|---------------|-----------|-------|
-|`druid.oss.accessKey`|the `AccessKey ID` of your account which can be used to access the bucket| |Must be set.|
-|`druid.oss.secretKey`|the `AccessKey Secret` of your account which can be used to access the bucket| |Must be set. |
-|`druid.oss.endpoint`|the endpoint url of your OSS storage| |Must be set.|
+After installation, add this `aliyun-oss-extensions` extension to `druid.extensions.loadList` in common.runtime.properties and then restart middle manager and historical nodes.
 
-if you want to use OSS as deep storage, use the configurations below
+## Configuration
 
-|Property|Description|Possible Values|Default|
-|--------|---------------|-----------|-------|
-|`druid.storage.type`| Global deep storage provider. Must be set to `oss` to make use of this extension. | oss |Must be set.|
-|`druid.storage.oss.bucket`|storage bucket name.| | Must be set.|
-|`druid.storage.oss.prefix`|a prefix string prepended to the file names for the segments published to aliyun OSS deep storage| druid/segments | |
+First add following OSS configurations to common.runtime.properties
 
-To save index logs to OSS, apply the configurations below:
+|Property|Description|Required|
+|--------|---------------|-----------|
+|`druid.oss.accessKey`|the `AccessKey ID` of your account which can be used to access the OSS bucket|yes|
+|`druid.oss.secretKey`|the `AccessKey Secret` of your account which can be used to access the OSS bucket| yes|
+|`druid.oss.endpoint`|the endpoint url of your OSS storage. <br/>If your Druid cluster is also hosted in the same region on Alibaba Cloud as the region of your OSS bucket, it's recommended to use the internal network endpoint url, so that any inbound and outbound traffic to the OSS bucket is free of charge. | yes|
```

Comment:
```suggestion
|`druid.oss.endpoint`|The endpoint URL of your OSS storage. <br/>If your Druid cluster is also hosted in the same region on Alibaba Cloud as the region of your OSS bucket, it's recommended to use the internal network endpoint url, so that any inbound and outbound traffic to the OSS bucket is free of charge. | yes|
```